### ISSUE TITLE
github actions fails

### ISSUE THREAD

**Issue by andreas-paul** [jdoiro3/mkdocs-multirepo-plugin#33]
Hi there,

thanks for providing this plugin for MkDocs! I've been trying to configure it but unfortunately I can't get it to work properly. Specifically, I am receiving the following error, when trying to clone a private repository in a Github action workflow:

```
mkdocs_multirepo_plugin.util.GitException: 
Cloning into 'project'...
fatal: could not read Username for 'https://github.com/': No such device or address
```
What I have tried so far is to add a personal access token from my account in github actions like so:

```
name: deploy to github pages
on:
  push:
    branches:
      - master 
      - main
jobs:
  deploy:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: 3.x
      - run: pip install mkdocs-material mkdocs-autolinks-plugin mkdocs-multirepo-plugin python-slugify
      - run: mkdocs gh-deploy --force
        env:
            AccessToken: ${{ secrets.docs_build_access }}
```

I have also tried to set environment variables directly for the private repo, to no avail. I'm at a loss here about how to configure github to enable access from one repo to the other, all of them private.  Any ideas or help would be appreciated!

**Comment by mikedeischle** [comment]
Hi,

I'm facing the same issue and tried multiple things, but without success.
E.g. added a git config to the pipeline
```   
git config --global url."https://gitlab-ci-token:${CI_JOB_TOKEN}@<gitlab-url>".insteadOf "https://<gitlab-url>"
git config --global url."https://gitlab-ci-token:${CI_JOB_TOKEN}@<gitlab-url>".insteadOf "ssh://git@<gitlab-url>"
git config --global url."https://gitlab-ci-token:${CI_JOB_TOKEN}@<gitlab-url>/".insteadOf "git@<gitlab-url>:"
```
Also setting the env variable `AccessToken` doesn't help here.

Locally with ssh key it works fine, so maybe adding this could be a solution?

**Comment by jdoiro3** [comment]
@andreas-paul and @mikedeischle, can you both remove the `AccessToken` and use this action instead: [ssh-agent](https://github.com/webfactory/ssh-agent). Let me know if it works.

**Comment by andreas-paul** [comment]
Thanks, I will try both solutions. In the meantime, I wrote a Github action script that basically does the same thing. Together with the [Awesome page plugin](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin), works nicely as well.

**Comment by achton** [comment]
@andreas-paul Can you elaborate on how you made it work? I'm looking to authenticate with private GitHub repos, and getting the username prompt when running `mkdocs build` (using the multirepo plugin as well).

**Comment by jdoiro3** [comment]
@andreas-paul, can you please confirm that this is resolved in version `v0.4.11`? See the README [here](https://github.com/jdoiro3/mkdocs-multirepo-plugin#github-actions) for the changes.

**Comment by jdoiro3** [comment]
Merged PR https://github.com/jdoiro3/mkdocs-multirepo-plugin/pull/60 should have resolved this. @wday-cs, feel free to continue this thread or open a new Issue if you still can't get it working.
